samcli/lib/observability/observability_info_puller.py

```diff
@@ -188,7 +191,7 @@ def tail(self, start_time: Optional[datetime] = None, filter_pattern: Optional[s
         except KeyboardInterrupt:
             LOG.info(" CTRL+C received, cancelling...")
             for puller in self._pullers:
-                puller.cancelled = True
+                puller.stop_tailing()
```

Comment:
We don’t need this for loop anymore, you can call `self.stop_tailing()` which will do exactly same thing.